src/test/java/unit/GcsBucketLifecycle.java

```diff
@@ -0,0 +1,390 @@
+package unit;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import bio.terra.cli.serialization.userfacing.UFAuthStatus;
+import com.google.api.client.util.DateTime;
+import com.google.cloud.Identity;
+import com.google.cloud.storage.Bucket;
+import com.google.cloud.storage.BucketInfo;
+import com.google.cloud.storage.StorageClass;
+import harness.TestCommand;
+import harness.baseclasses.SingleWorkspaceUnit;
+import harness.utils.ExternalGCSBuckets;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+/** Tests for specifying lifecycle rules for controlled GCS buckets. */
+@Tag("unit")
+public class GcsBucketLifecycle extends SingleWorkspaceUnit {
+  // external bucket to use for testing the JSON format against GCS directly
+  private Bucket externalBucket;
+
+  @Override
+  @BeforeAll
+  protected void setupOnce() throws IOException {
+    super.setupOnce();
+    externalBucket = ExternalGCSBuckets.createBucket();
+
+    // `terra auth status --format=json`
+    UFAuthStatus authStatus =
+        TestCommand.runAndParseCommandExpectSuccess(UFAuthStatus.class, "auth", "status");
+
+    // grant the user's proxy group write access to the bucket, so we can test calling `terra gsutil
+    // lifecycle` with the same JSON format used for creating controlled bucket resources with
+    // lifecycle rules
+    ExternalGCSBuckets.grantWriteAccess(externalBucket, Identity.group(authStatus.proxyGroupEmail));
```

Comment:
nit: since this is only used once in a single test, maybe just do all of this setup/tear down in that test?

Reply:
I want the teardown to happen regardless of whether the test passes though. I could put just the setup in the test method, and then do the teardown in the `@AfterAll` method -- that's what I do for SAM groups, for example. But since we only need one, it seemed more readable to put setup/cleanup in `@BeforeAll` and `@AfterAll`. I think if we get a test that creates multiple external buckets, then following the pattern of `SamGroups`, which tracks groups and then provides a method for deleting them all at the end, would be good.